File: LAB_2/Tablica.java
public class Tablica {
    private int[] dane;

    // Konstruktor domyślny
    public Tablica() {
        this.dane = new int[10];
        System.out.println("Domyślny konstruktor: alokowano tablicę o rozmiarze 10.");
    }

    // Konstruktor z parametrem
    public Tablica(int rozmiar) {
        this.dane = new int[rozmiar];
        System.out.println("Konstruktor z parametrem: alokowano tablicę o rozmiarze " + rozmiar + ".");
    }

    public static void main(String[] args) {
        // Testowanie klasy Tablica
        Tablica t1 = new Tablica();
        Tablica t2 = new Tablica(20);
    }
}

File: LAB_2/tablica.cpp
#include <iostream>

using namespace std;

class Tablica {
private:
    int* dane;
    int rozmiar;

public:
    // Konstruktor domyślny
    Tablica() : rozmiar(10) {
        dane = new int[rozmiar];
        cout << "Domyślny konstruktor: alokowano tablicę o rozmiarze " << rozmiar << ".\n";
    }

    // Konstruktor z parametrem
    Tablica(int r) : rozmiar(r) {
        dane = new int[rozmiar];
        cout << "Konstruktor z parametrem: alokowano tablicę o rozmiarze " << rozmiar << ".\n";
    }

    // Destruktor
    ~Tablica() {
        delete[] dane;
        cout << "Destruktor: zwolniono pamięć dla tablicy o rozmiarze " << rozmiar << ".\n";
    }
};

int main() {
    Tablica t1;
    Tablica t2(20);

    return 0;
}
